component/src/main/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestAuthenticator.java

```diff
@@ -0,0 +1,394 @@
+		} catch (OAuthSystemException e) {
+			throw new AuthenticationFailedException("Error while generating the OAuthClientRequest", e);
+		}
+	}
+
+	/**
+	 * This method is to get the CallBackURL.
+	 */
+	@Override
+	protected String getCallbackUrl(Map<String, String> authenticatorProperties) {
+		return authenticatorProperties.get(PinterestAuthenticatorConstants.CALLBACK_URL);
+	}
+
+	/**
+	 * This method are overridden for extra claim request to Pinterest end-point.
+	 *
+	 * @param request  the http request
+	 * @param response the http response
+	 * @param context  the authentication context
+	 * @throws AuthenticationFailedException
+	 */
+	@Override
+	protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+	                                             AuthenticationContext context) throws AuthenticationFailedException {
+		try {
+			Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+			String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+			String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+			String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+			String callbackUrl = getCallbackUrl(authenticatorProperties);
+			OAuthAuthzResponse authzResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+			String code = authzResponse.getCode();
+			OAuthClientRequest accessRequest;
+			try {
+				accessRequest =
+						OAuthClientRequest.tokenLocation(tokenEndPoint).setGrantType(GrantType.AUTHORIZATION_CODE)
+						                  .setClientId(clientId).setClientSecret(clientSecret)
+						                  .setRedirectURI(callbackUrl).setCode(code).buildBodyMessage();
+				// create OAuth client that uses custom http client under the hood
+				OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+				OAuthClientResponse oAuthResponse;
+				oAuthResponse = oAuthClient.accessToken(accessRequest);
+				String accessToken = oAuthResponse.getParam(PinterestAuthenticatorConstants.ACCESS_TOKEN);
+				if (StringUtils.isNotEmpty(accessToken)) {
+					Map<ClaimMapping, String> claims = buildClaims(oAuthResponse, authenticatorProperties);
+					if (claims != null && !claims.isEmpty()) {
+						//Find the subject from the IDP claim mapping, subject Claim URI.
+						String subjectFromClaims = FrameworkUtils
+								.getFederatedSubjectFromClaims(context.getExternalIdP().getIdentityProvider(), claims);
+						associateSubjectFromClaims(context, subjectFromClaims, claims);
+					} else {
+						throw new AuthenticationFailedException("Claims for the user not found for access Token : " +
+						                                        accessToken);
+					}
+				} else {
+					throw new AuthenticationFailedException("Authentication Failed, access token is not available");
```

Comment:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-pinterest/pull/1/commits/9677bf19d79c461d484e61e2bf3c9332c6305ef6